sdks/java/core/src/main/java/org/apache/beam/sdk/io/WriteFiles.java

```diff
@@ -686,74 +678,65 @@ public void processElement(ProcessContext context) throws IOException {
     } catch (CannotProvideCoderException | NonDeterministicException e) {
       throw new RuntimeException(e);
     }
+    FileResultCoder<DestinationT> fileResultCoder =
+        FileResultCoder.of(shardedWindowCoder, destinationCoder);
 
+    PCollection<FileResult<DestinationT>> results;
     if (computeNumShards == null && numShardsProvider == null) {
-      numShardsView = null;
       TupleTag<FileResult<DestinationT>> writtenRecordsTag =
           new TupleTag<>("writtenRecordsTag");
-      TupleTag<KV<ShardedKey<Integer>, UserT>> unwrittedRecordsTag =
-          new TupleTag<>("unwrittenRecordsTag");
+      TupleTag<KV<ShardedKey<Integer>, UserT>> spilledRecordsTag =
+          new TupleTag<>("spilledRecordsTag");
```

Comment:
changing the TupleTag name breaks update compatibility. Update compatibility is the whole reason we used an explicit name here in the first place.